demos/demo-modules/selectable-period/component.ts

```diff
@@ -34,22 +34,30 @@ export class DemoComponent {
 
   events: CalendarEvent[] = [];
 
+  selectedDays: any = [];
+  hexTimeArray: any = [];
+
   dayClicked(day: CalendarMonthViewDay): void {
-    if (this.selectedMonthViewDay) {
+    this.selectedMonthViewDay = day;
+    const selectedDateTime = this.selectedMonthViewDay.date.getTime();
+    const dateIndex = this.hexTimeArray.indexOf(selectedDateTime);
+    if (dateIndex > -1) {
       delete this.selectedMonthViewDay.cssClass;
+      this.selectedDays.splice(dateIndex, 1);
+      this.hexTimeArray.splice(dateIndex, 1);
+    } else {
+      this.selectedDays.push(this.selectedMonthViewDay);
+      this.hexTimeArray.push(selectedDateTime);
+      day.cssClass = 'cal-day-selected';
+      this.selectedMonthViewDay = day;
     }
-    day.cssClass = 'cal-day-selected';
-    this.selectedMonthViewDay = day;
   }
 
   beforeMonthViewRender({ body }: { body: CalendarMonthViewDay[] }): void {
     body.forEach(day => {
-      if (
-        this.selectedMonthViewDay &&
-        day.date.getTime() === this.selectedMonthViewDay.date.getTime()
-      ) {
+      const index = this.hexTimeArray.indexOf(day.date.getTime());
+      if (index > -1) {
```

Comment:
Same here as well, you can simplify this code a bit by only using `selectedDays` and here do:
```
if (this.selectedDays.some(selectedDay => selectedDay.date.getTime() === day.date.getTime())) {
```